### PR TITLE
Add Remove Operation

### PR DESCRIPTION
**Pull request by denis-rossati**
## Summary

- The new operation has been added to activeRecord.
- The new operation has been added to patch file.
- The new operation has been added to operationSchemas file.

---

I maded/updated the tests to:

- Check if a chained operation with the new operator generates a record of operations.
- Check if the operation does not throw anything when it's completed.
- Check if the operation throw a error and certain message when the path is invalid
- Check if the operation throw an error and a message when the value is not declared.

--- 

### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas `(there's no hard-to-understand areas)`
- [ ] I have made corresponding changes to the documentation `(I'll do it in plug.js)`
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules `(I'm don't know how to be sure that this task is completed)`
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by Fryuni** [comment]
> Any dependent changes have been merged and published in downstream modules (I'm don't know how to be sure that this task is completed)
This check is to remind you to check that this PR does not depend on pending PRs from other repos. In this case it doesn't

Had you already opened the PR on the Plug to update the docs you should not check this box there yet since it depends on this PR here

😄

File: test/schemas/operationSchemas.test.ts [hunk 1]
```diff
@@ -352,6 +353,44 @@ describe('A decrement operation schema', () => {
...
+describe('A remove operation schema', () => {
+    test.each([
+        [1],
+        [1.5],
+        ['foo'],
+        [true],
+        [simpleArray],
+        [simpleMap],
+        [complexMap],
+    ])('should remove value %s', (value: any) => {
```
Comment (amorimjuliana) [inline test/schemas/operationSchemas.test.ts:365]: The pattern here is to use `should allow value %s` for the test name.

File: test/schemas/operationSchemas.test.ts [hunk 2]
```diff
@@ -352,6 +353,44 @@ describe('A decrement operation schema', () => {
...
+    ])('should fail if path is invalid', (path: string) => {
+        function validate(): void {
+            removeOperation.validate({path: path, value: 'bar'});
+        }
+
+        expect(validate).toThrow(Error);
+        expect(validate).toThrow('Invalid pointer format at path \'/path\'.');
+    });
+
+    test('should fail if value is invalid', () => {
```
Comment (amorimjuliana) [inline test/schemas/operationSchemas.test.ts:385]: And here, `should not allow value %s`.

File: test/schemas/operationSchemas.test.ts [hunk 3]
```diff
@@ -352,6 +353,44 @@ describe('A decrement operation schema', () => {
...
+
+        expect(validate).not.toThrow(Error);
+    });
+
+    test.each([
+        ['foo.'],
+        [''],
+    ])('should fail if path is invalid', (path: string) => {
+        function validate(): void {
+            removeOperation.validate({path: path, value: 'bar'});
```
Comment (amorimjuliana) [inline test/schemas/operationSchemas.test.ts:378]: To keep all tests standardized, please add the empty case on others as well, please.

File: test/schemas/operationSchemas.test.ts (hunk 2 above)
Comment (amorimjuliana) [reply test/schemas/operationSchemas.test.ts:385]: Also, please rely on other tests examples for the invalid cases.

**Review by amorimjuliana — CHANGES_REQUESTED** [review]

**Review by amorimjuliana — APPROVED** [review]

**Review by marcospassos — APPROVED** [review]

**Review by marcospassos — CHANGES_REQUESTED** [review]
Please add documentation for the new operation.

**Pull request merged**